src/components/Tables/Filters.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { styled } from "styled-components";
+import { TableHeaders } from "@/components/Tables/Table";
+
+export interface Filter<Data, State> {
+    shouldFilter(data: Data, state: State): boolean;
+    filterComponent(state: State, setState: (state: State) => void): React.ReactElement;
+    state: State;
+    initialState: State;
+    clear(state: State, setState: (state: State) => void): void;
+}
+
+export interface KeyedFilter<Data, N extends keyof Data, State> extends Filter<Data, State> {
+    key: N;
+    label: string;
+}
+
+const keyedFilter = <Data, N extends keyof Data, State>(
+    key: N,
+    label: string,
+    filter: Filter<Data, State>
+): KeyedFilter<Data, N, State> => {
+    return {
+        key,
+        label,
+        ...filter,
+    };
+};
+
+export const headerLabelFromKey = <Data, N extends keyof Data>(
+    headers: TableHeaders<Data>,
+    key: N
+): string => {
+    return headers.find(([headerKey]) => headerKey === key)![1];
+};
+
+const StyledFilterBar = styled.input`
```

Comment:
Any reason for using the normal input not FreeFormTextInput?

Reply:
I can't get it to look consistent, and this is what it always was